### ISSUE TITLE
SIGSEGV installing a game

### ISSUE THREAD

**Issue by neuromancer** [tkashkin/GameHub#232]
Using the latest dev revision (cbac77743def348e58c20ee2b6336ab9e74f9c4c), it crashes when I try to install some games like "Magic Carpet". Backtrace is here:

```
Thread 1 "com.github.tkas" received signal SIGSEGV, Segmentation fault.
0x000055555557f843 in game_hub_data_runnable_installer_ref (instance=0x0) at ../src/data/Runnable.vala:321
321			public abstract class Installer
(gdb) bt
#0  0x000055555557f843 in game_hub_data_runnable_installer_ref (instance=0x0) at ../src/data/Runnable.vala:321
#1  0x00005555555b3fcc in game_hub_data_sources_gog_gog_game_installer_real_fetch_parts_co (_data_=0x555555bb00b0)
    at ../src/data/sources/gog/GOGGame.vala:586
#2  0x00005555555b35f3 in _game_hub_data_sources_gog_gog_game_installer_real_fetch_parts_co_gsource_func (self=0x555555bb00b0)
    at GOGGame.c:7885
#3  0x00007ffff7021661 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#4  0x00007ffff7023739 in  () at /usr/lib/libglib-2.0.so.0
#5  0x00007ffff702377e in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#6  0x00007ffff71d9f5e in g_application_run () at /usr/lib/libgio-2.0.so.0
#7  0x000055555556b7be in game_hub_application_main (args=0x7fffffffe7a8, args_length1=2) at ../src/app.vala:219
#8  0x000055555556b815 in main (argc=2, argv=0x7fffffffe7a8) at ../src/app.vala:204
(gdb) x/i $rip
=> 0x55555557f843 <game_hub_data_runnable_installer_ref+24>:	lock addl $0x1,(%rax)
(gdb) info registers 
rax            0x8                 8
rbx            0x55555ca086a0      93825114605216
rcx            0x55555ca086a0      93825114605216
rdx            0x555555bb00b0      93824998899888
rsi            0x0                 0
rdi            0x0                 0
rbp            0x7fffffffe470      0x7fffffffe470
rsp            0x7fffffffe470      0x7fffffffe470
r8             0x7ffff7fcf0b0      140737353937072
r9             0x7ffff70da300      140737338254080
r10            0x68260             426592
r11            0x476               1142
r12            0x7fffd401a180      140736750264704
r13            0x0                 0
r14            0x7fffd405a470      140736750527600
r15            0x7ffff701bd60      140737337474400
rip            0x55555557f843      0x55555557f843 <game_hub_data_runnable_installer_ref+24>
eflags         0x10202             [ IF RF ]
cs             0x33                51
ss             0x2b                43
ds             0x0                 0
es             0x0                 0
fs             0x0                 0
gs             0x0                 0

```

**Comment by tkashkin** [comment]
It should probably be fixed in abc24ad.

**Comment by neuromancer** [comment]
No more crashes. Thanks a lot!
